packages/rest/src/sequence.ts

```diff
@@ -121,3 +122,83 @@ export class DefaultSequence implements SequenceHandler {
     }
   }
 }
+
+/**
+ * A sequence implementation using middleware chains
+ */
+export class MiddlewareSequence implements SequenceHandler {
+  static defaultOptions: InvokeMiddlewareOptions = {
+    chain: RestTags.REST_MIDDLEWARE_CHAIN,
+    orderedGroups: [
+      // Please note that middleware is cascading. The `SendResponse` is
+      // added first to invoke downstream middleware to get the result or
+      // catch errors so that it can produce the http response.
+      'sendResponse',
```

Comment:
If these are actually groups, can we use constants instead of string names here?